trick_source/sim_services/DataRecord/DataRecordGroup.cpp

```diff
@@ -339,6 +340,24 @@ int Trick::DataRecordGroup::add_change_variable( std::string in_name ) {
 
 }
 
+bool Trick::DataRecordGroup::isSupportedType(REF2 * ref2, std::string& message) {
+    if (ref2->attr->type == TRICK_STRING || ref2->attr->type == TRICK_STL || ref2->attr->type == TRICK_STRUCTURED) {
+        message = "Cannot Data Record variable " + std::string(ref2->reference) + " of unsupported type " + std::to_string(ref2->attr->type);
+        return false;
+    }
+
+    const static std::regex array_element_regex(".+(?:\\[\\d+\\])+$");
```

Comment:
This will only match if the string ends with [num]. I am concerned that I could be missing some cases here that will cause unintended consequences. Another option is to just let this case go completely.

Reply:
Found the actual way to do this from varserver code.